File: cmd_marriage.py
# cmd_marriage.py
import random
from db_manager import users_col, get_user

# ======= إعداد الزواج =======
marriage_enabled = True

async def is_already_married(uid):
    marriage = await users_col.find_one({
        "$or": [{"husband_uid": uid}, {"wife_uid": uid}]
    })
    return marriage is not None

async def get_marriage(uid):
    return await users_col.find_one({
        "$or": [{"husband_uid": uid}, {"wife_uid": uid}]
    })

# ======= أمر زوجني =======
async def marry_user(bot, message, target_identifier=None):
    user = await get_user(message.from_user.id, message.from_user.first_name)

    if await is_already_married(user["uid"]):
        await bot.send_message(message.chat.id, f"⚠️ {user['name']} أنت متزوج بالفعل!")
        return

    target_user = None

    # ===== إذا رد على شخص =====
    if message.reply_to_message:
        target_user = await get_user(message.reply_to_message.from_user.id, message.reply_to_message.from_user.first_name)
    # ===== إذا ذكر UID أو @username =====
    elif target_identifier:
        if target_identifier.startswith("@"):
            target_user = await users_col.find_one({"username": target_identifier[1:]})
        else:
            try:
                target_uid = int(target_identifier)
                target_user = await get_user(target_uid)
            except:
                await bot.send_message(message.chat.id, "⚠️ الرجاء كتابة UID صالح أو @username")
                return
    # ===== بدون أي تحديد =====
    if not target_user:
        all_users = await users_col.find({"uid": {"$ne": user["uid"]}}).to_list(length=None)
        if not all_users:
            await bot.send_message(message.chat.id, "⚠️ لا يوجد مستخدمين آخرين في قاعدة البيانات!")
            return
        target_user = random.choice(all_users)

    if await is_already_married(target_user["uid"]):
        await bot.send_message(message.chat.id, f"⚠️ {target_user['name']} هذا الشخص متزوج بالفعل!")
        return

    # تسجيل الزواج
    await users_col.insert_one({
        "husband_uid": user["uid"],
        "wife_uid": target_user["uid"],
        "married_at": int(message.date.timestamp())
    })

    text = f"💍 تم الزواج بنجاح بين:\n• {user['name']} ❤️ {target_user['name']}"
    meme_url = "https://i.imgur.com/9bX5YUw.jpg"
    await bot.send_photo(message.chat.id, meme_url, caption=text)

# ======= أمر طلقني =======
async def divorce(bot, message):
    user = await get_user(message.from_user.id, message.from_user.first_name)
    marriage = await get_marriage(user["uid"])
    if not marriage:
        await bot.send_message(message.chat.id, f"⚠️ {user['name']} أنت غير متزوج حالياً!")
        return

    await users_col.delete_one({"_id": marriage["_id"]})
    await bot.send_message(message.chat.id, f"💔 {user['name']} تم الطلاق بنجاح!")

# ======= قائمة المتزوجين =======
async def list_married(bot, message):
    all_marriages = await users_col.find({"husband_uid": {"$exists": True}}).to_list(length=None)
    if not all_marriages:
        await bot.send_message(message.chat.id, "لا يوجد متزوجين حالياً.")
        return

    text = "💑 قائمة المتزوجين:\n"
    for m in all_marriages:
        husband = await get_user(m["husband_uid"])
        wife = await get_user(m["wife_uid"])
        text += f"• {husband['name']} ❤️ {wife['name']}\n"

    await bot.send_message(message.chat.id, text)

# ======= Handler =======
async def handle(bot, message):
    if not marriage_enabled:
        return

    text = message.text
    if not text:
        return

    parts = text.strip().split()
    command = parts[0]
    target_identifier = parts[1] if len(parts) > 1 else None

    if command == "زوجني":
        await marry_user(bot, message, target_identifier)
    elif command == "طلقني":
        await divorce(bot, message)
    elif command == "قائمة المتزوجين":
        await list_married(bot, message)

File: cmd_marriage.js
// ملف: cmd_marriage.js
const COMMANDS = ["زوجني", "طلقني", "قائمة المتزوجين"];
const db_manager = require("./db_manager");

// ======= إعداد الزواج =======
let marriage_enabled = true;

async function isAlreadyMarried(uid) {
    const marriage = await db_manager.users.findOne({
        $or: [{ husband_uid: uid }, { wife_uid: uid }]
    });
    return marriage !== null;
}

async function getMarriage(uid) {
    return await db_manager.users.findOne({
        $or: [{ husband_uid: uid }, { wife_uid: uid }]
    });
}

// ======= أمر زوجني =======
async function marryUser(bot, message, target_identifier) {
    const user = await db_manager.getUser(message.from.id);

    if (await isAlreadyMarried(user.uid)) {
        bot.replyTo(message, "⚠️ أنت متزوج بالفعل!");
        return;
    }

    let target_user = null;

    if (!target_identifier) {
        // اختيار عشوائي من المستخدمين الموجودين في DB
        const allUsers = await db_manager.users.find({ uid: { $ne: user.uid } }).toArray();
        target_user = allUsers[Math.floor(Math.random() * allUsers.length)];
    } else if (target_identifier.startsWith("@")) {
        target_user = await db_manager.users.findOne({ username: target_identifier.slice(1) });
    } else {
        try {
            const target_uid = parseInt(target_identifier);
            target_user = await db_manager.getUser(target_uid);
        } catch {
            bot.replyTo(message, "⚠️ الرجاء كتابة UID صالح أو @username");
            return;
        }
    }

    if (!target_user) {
        bot.replyTo(message, "⚠️ لم يتم العثور على المستخدم المطلوب.");
        return;
    }

    if (await isAlreadyMarried(target_user.uid)) {
        bot.replyTo(message, "⚠️ هذا الشخص متزوج بالفعل!");
        return;
    }

    // تسجيل الزواج
    await db_manager.users.insertOne({
        husband_uid: user.uid,
        wife_uid: target_user.uid,
        married_at: Date.now()
    });

    const text = `💍 تم الزواج بنجاح بين:
• ${user.uid} (${user.username || "بدون اسم"}) ❤️
• ${target_user.uid} (${target_user.username || "بدون اسم"})`;

    // إرسال صورة ميمز ثابتة مع النص (يمكن تغيير الرابط حسب اختيارك)
    const meme_url = "https://i.imgur.com/9bX5YUw.jpg"; // مثال لميمز زواج
    bot.sendPhoto(message.chat.id, meme_url, { caption: text });
}

// ======= أمر طلقني =======
async function divorce(bot, message) {
    const user = await db_manager.getUser(message.from.id);
    const marriage = await getMarriage(user.uid);
    if (!marriage) {
        bot.replyTo(message, "⚠️ أنت غير متزوج حالياً!");
        return;
    }

    await db_manager.users.deleteOne({ _id: marriage._id });
    bot.replyTo(message, "💔 تم الطلاق بنجاح!");
}

// ======= قائمة المتزوجين =======
async function listMarried(bot, message) {
    const all_marriages = await db_manager.users.find({ husband_uid: { $exists: true } }).toArray();
    if (!all_marriages.length) {
        bot.replyTo(message, "لا يوجد متزوجين حالياً.");
        return;
    }

    let text = "💑 قائمة المتزوجين:\n";
    for (const m of all_marriages) {
        const husband = await db_manager.getUser(m.husband_uid);
        const wife = await db_manager.getUser(m.wife_uid);
        text += `• ${husband.uid} (${husband.username || "بدون اسم"}) ❤️ ${wife.uid} (${wife.username || "بدون اسم"})\n`;
    }

    bot.sendMessage(message.chat.id, text);
}

// ======= Handler =======
async function handle(bot, message) {
    if (!marriage_enabled) return;

    const text = message.text?.trim();
    if (!text) return;

    if (text.startsWith("زوجني")) {
        const parts = text.split(/\s+/);
        const target_identifier = parts[1] || null; // إذا ما ذكر، يكون عشوائي
        await marryUser(bot, message, target_identifier);
    } else if (text === "طلقني") {
        await divorce(bot, message);
    } else if (text === "قائمة المتزوجين") {
        await listMarried(bot, message);
    }
}

module.exports = { handle };
